Kerberos.NET/Client/Krb5CredentialCache.cs

```diff
@@ -0,0 +1,600 @@
+            }
+
+            this.ReadHeader(buffer);
+            this.DefaultPrincipalName = ReadPrincipal(buffer);
+
+            this.ReadCredentials(buffer);
+        }
+
+        internal byte[] Serialize()
+        {
+            var buffer = new NdrBuffer(align: false);
+
+            buffer.WriteByte(Magic);
+            buffer.WriteByte(ExpectedVersion);
+
+            this.WriteHeader(buffer);
+            WritePrincipal(this.DefaultPrincipalName, buffer);
+
+            this.WriteCredentials(buffer);
+
+            return buffer.ToMemory(0).ToArray();
+        }
+
+        internal void Add(TicketCacheEntry entry)
+        {
+            var existing = FindCredential(entry.Key);
+
+            if (existing != null)
+            {
+                this.Credentials.Remove(existing);
+            }
+
+            if (entry.Value is KerberosClientCacheEntry entryValue)
+            {
+                KrbEncryptionKey sessionKey;
+
+                if (entryValue.KdcResponse is KrbAsRep asRep)
+                {
+                    sessionKey = entryValue.SessionKey;
+
+                    if (!this.Credentials.Any())
+                    {
+                        this.DefaultPrincipalName = FromResponse(asRep, asRep.CName);
+                    }
+                }
+                else
+                {
+                    var encKdcRepPart = entryValue.KdcResponse.EncPart.Decrypt(
+                        entryValue.SessionKey.AsKey(),
+                        entryValue.SessionKey.Usage,
+                        d => KrbEncTgsRepPart.DecodeApplication(d)
+                    );
+
+                    sessionKey = encKdcRepPart.Key;
+                }
+
+                this.Credentials.Add(new Krb5Credential
+                {
+                    Ticket = entryValue.KdcResponse.Ticket.EncodeApplication(),
+                    KeyBlock = new KeyValuePair<EncryptionType, ReadOnlyMemory<byte>>(sessionKey.EType, sessionKey.KeyValue),
+                    Client = FromResponse(entryValue.KdcResponse, entryValue.KdcResponse.CName),
+                    Server = FromResponse(entryValue.KdcResponse, entryValue.KdcResponse.Ticket.SName),
+                    AuthData = new List<KrbAuthorizationData>(),
+                    EndTime = entry.Expires,
+                    RenewTill = entry.RenewUntil ?? DateTimeOffset.MinValue,
+                    Addresses = new List<KrbHostAddress>(),
+                    SecondTicket = Array.Empty<byte>()
+                });
+            }
+        }
+
+        private static PrincipalName FromResponse(KrbKdcRep asRep, KrbPrincipalName name)
+        {
+            return new PrincipalName(asRep.CName.Type, asRep.CRealm, name.Name);
+        }
+
+        internal object GetCacheItem(string key)
+        {
+            Krb5Credential cred = FindCredential(key);
+
+            if (cred == null)
```

Comment:
```suggestion
            if (cred is null)
```
`is null` is a "direct" null-check.
`== null` may need more runtime-code iif the `==` operator is defined explicitely.
Here it may not be a point, but it's good hygiene in dotnet-code.